packages/cli/src/commands/hydrogen/deploy.ts

```diff
@@ -73,11 +74,19 @@ export default class Deploy extends Command {
       required: false,
       default: false,
     }),
+    'custom-build': Flags.boolean({
```

Comment:
I this this could be a little inflexible, no? What if you need to pass arguments to this command, too? What do you think about passing changing this to be a build command that you passing as a string. That means we could skip dealing with `getPackageManager` logic and run whatever the user provides.

```sh
h2 deploy --build "FOO=1 BAR=example.com h2 build --no-lockfile-check --no-sourcemap"
```

Reply:
I like this. Then you could specify `yarn build` here and it makes more sense with the default option not using package.json at all.

Reply:
My thinking was users would get that flexibility all the same, as they could add those flags to their build script in package.json.

But letting them supply the command entirely is fine too, simplifies the code, and like Jason said, fits in. with not using package.json at all. I've named it `build-command` though 😄 .

Reply:
Cool! I'll give this a proper 🎩 next week!